pennylane/math/tf_box.py

```diff
@@ -68,6 +68,26 @@ def __len__(self):
     def astensor(tensor):
         return tf.convert_to_tensor(tensor)
 
+    @staticmethod
+    @wrap_output
+    def block_diag(values):
+        tensors = TensorFlowBox.unbox_list(values)
+        tensors = TensorFlowBox._coerce_types(tensors)
+        int_dtype = None
+
+        if tensors[0].dtype in (tf.int32, tf.int64):
+            int_dtype = tensors[0].dtype
+            tensors = [tf.cast(t, tf.float32) for t in tensors]

```

Comment:
The casting feels a little awkward. I assume this is a TF limitation?

Reply:
Yeah, it's super weird. I discovered while writing this that TF only allows creating block diagonal matrices from float/complex tensors, _not_ integer tensors 🙁

Reply:
It's very unlikely the metric tensor will ever be integral, but at the same time, the `qml.math` should try and be as consistent across interfaces as possible to avoid future issues/bugs.

Reply:
Easiest way to do that is to check if the type is integral and raise an error if so since we don't expect it to ever happen. If we ever hit it then we deal with it.